File: backend/routes/tasksRoutes.js
/**
 * routes/tasks.js
 * ---------------
 * Route definitions for the Task CRUD API.
 * All logic lives in taskController — this file is routing only.
 *
 * Mounted at: /api/tasks
 *
 * ┌──────────────────────────────────────────────────────────────────┐
 * │ Method  │ Path            │ Action                               │
 * ├──────────────────────────────────────────────────────────────────┤
 * │ POST    │ /               │ Create tasks from a list of titles   │
 * │ GET     │ /               │ Get all tasks (with optional filters)│
 * │ GET     │ /:id            │ Get one task by id                   │
 * │ PATCH   │ /:id            │ Update title / isCompleted           │
 * │ DELETE  │ /:id            │ Delete one task                      │
 * │ DELETE  │ /               │ Delete ALL tasks                     │
 * └──────────────────────────────────────────────────────────────────┘
 */

const express = require('express');
const router  = express.Router();
const {
  createTasks,
  getAllTasks,
  getTaskById,
  updateTask,
  deleteTask,
  deleteAllTasks,
} = require('../controllers/taskController');

// ── Collection routes (/api/tasks) ───────────────────────────────────────────
router.post  ('/',    createTasks);
router.get   ('/',    getAllTasks);
router.delete('/',    deleteAllTasks);

// ── Individual resource routes (/api/tasks/:id) ──────────────────────────────
router.get   ('/:id', getTaskById);
router.patch ('/:id', updateTask);
router.delete('/:id', deleteTask);

module.exports = router;

File: backend/allFastAPI/routers/tasksRoutes.py
from fastapi import APIRouter, status
from typing import List, Optional

# Assuming your controller logic is in a file named taskController.py
from controllers import taskController

router = APIRouter()

# ── Collection routes (/api/tasks) ──

@router.post("/", status_code=status.HTTP_201_CREATED)
async def create_tasks(payload: List[str]):
    """Create tasks from a list of titles"""
    return await taskController.create_tasks(payload)

@router.get("/")
async def get_all_tasks(completed: Optional[bool] = None):
    """Get all tasks (with optional filters)"""
    return await taskController.get_all_tasks(completed)

@router.delete("/")
async def delete_all_tasks():
    """Delete ALL tasks"""
    return await taskController.delete_all_tasks()


# ── Individual resource routes (/api/tasks/{id}) ──

@router.get("/{task_id}")
async def get_task_by_id(task_id: str):
    """Get one task by id"""
    return await taskController.get_task_by_id(task_id)

@router.patch("/{task_id}")
async def update_task(task_id: str, update_data: dict):
    """Update title / isCompleted"""
    return await taskController.update_task(task_id, update_data)

@router.delete("/{task_id}")
async def delete_task(task_id: str):
    """Delete one task"""
    return await taskController.delete_task(task_id)
